internal/pkg/cli/env_show.go

```diff
@@ -74,9 +84,60 @@ func (o *showEnvOpts) Ask() error {
 
 // Execute shows the environments through the prompt.
 func (o *showEnvOpts) Execute() error {
+	if err := o.initDescriber(o); err != nil {
+		return err
+	}
+	env, err := o.retrieveData()
+	if err != nil {
+		return err
+	}
+	if o.shouldOutputJSON {
+		data, err := env.JSONString()
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(o.w, data)
+	} else {
+		fmt.Println(env)
+		//} else {
+		//	fmt.Fprintf(o.w, env.HumanString())
+	}
+
 	return nil
 }
 
+func (o *showEnvOpts) retrieveData() (*describe.EnvSummary, error) {
```

Comment:
What do you think of having all this logic be part of a `Describe` method in the `EnvDescriber` object?
This way the `Execute` method is dead simple:
```go
summary, err := o.describer.Describe()
if err != nil {
	return err
}
if o.shouldOutputJSON {
	...
} else {
	...
}
```

Reply:
Ah, yes, like in app_status. Gotcha. Thanks!

Reply:
@efekarakus the way `o.retrieveData()` is structured right now provides the same simplicity when called from the `Execute()` block. Since this PR is meant to mostly cover the initial interface definitions, I think it's alright to keep this as is for now and maybe just make a note to possibly refactor later. 

(Also worth noting that this is the structure used in both project_show and app_show).

Reply:
The `retrieveData` method is a bit awkward because it populates a struct that's defined in the `describe` package and not the `cli` package. Instead, when we push the work down to the `describe.EnvDescriber` struct it becomes the responsibility of the `describe` package and keeps the CLI's `Execute` method thin.

If we'd like to keep it as is to unblock, then I'd like the next PR to move it down just like `app_status.go` and a TODO with a issue link explaining the task. 
The other files: `project_show` and `app_show` are older and we have learned since then :)